### PR TITLE
Rest client option to not decode JSON to a python dict

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: lib/vsc/utils/rest.py [hunk 1]
```diff
@@ -193,10 +194,13 @@ def request(self, method, url, body, headers, content_type=None):
...
-                    pybody = json.loads(body)
-                except ValueError:
-                    pybody = body
+                if self.decode:
+                    try:
+                        pybody = json.loads(body)
+                    except ValueError:
+                        pybody = body
+                else:
+                    pybody=body
```
[the remark this answers is not in the document]
Comment (itkovian) [reply lib/vsc/utils/rest.py:203]: ah yes.,

**Comment by itkovian** [comment]
the use case is being able to convert it to a know data class instance.

**Comment by stdweird** [comment]
one that doesn't do a json.load?

**Comment by itkovian** [comment]
one that gets the json instead of a dict

**Comment by stdweird** [comment]
but what does the client do if not `json.loads`? (i'm trying to understand what the benefit could be)

**Comment by itkovian** [comment]
the client will convert the json to a data class instance, not to a generic python dict.

**Comment by stdweird** [comment]
i'll merge it, just curious how this will not introduce even more duplicate code

**Pull request merged**